core/templates/dev/head/pages/story-viewer-page/story-viewer-page.directive.ts

```diff
@@ -184,7 +181,31 @@ angular.module('oppia').directive('storyViewerPage', [
               if (!storyNodes[i].isCompleted() &&
                 (storyNodes[i].getId() !==
                 ctrl.storyPlaythroughObject.getNextPendingNodeId())) {
-                thumbnailColor = 'grey';
+                if (
+                  storyNodes[i].getExplorationSummaryObject(
+                  ).thumbnail_bg_color.length === 7) {
```

Comment:
Perhaps add a comment explaining the significance of 7, or this condition in general -- it's not clear to me.

Reply:
Oh, this was needed in a previous iteration where I modified thumbnailColor in place, don't think this is needed now, will remove it.